tests/subsys/settings/nvs/raw/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  system.settings.nvs:
+    platform_whitelist: qemu_x86
```

Comment:
sanitycheck is also used for local hardware testing using the `--device-testing` flag, and whitelists like this will prevent the test from being executed on hardware. We really need to avoid whitelists whenever possible

Reply:
okay, I see. If the test for qemu_86 is still picked up by the CI without this whitelist entry, I can get rid of it

Reply:
I removed the ```platform_whitelist``` attribute

Reply:
After I removed ```platform_whitelist```, the ```ztest.h``` file is not visible during compilation in the Zephyr CI tests. Is it related or do I need to rebase?

Reply:
Rebase didn't help.

I brought back this attribute since the Zephyr CI failed without having it defined in the yaml file.

Reply:
Please figure out why `ztest.h` isn't being found. `qemu_x86` isn't the only platform that runs tests

Reply:
Apparently, the Zephyr CI scripts require this attribute to prepare the environment for test suite execution. Could you give me an example of the test without ```platform_whitelist```?

Reply:
`tests/drivers/spi/spi_loopback/testcase.yaml`

Reply:
AFIK without specifying qemu we can't build test as a ztes header is missing.

Reply:
there is something wrong here. ztest has nothing to do with qemu_x86

Reply:
Yes, something is off. The test that @MaureenHelm referred to does not have a whitelist and also includes `ztest.h`:
https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/spi/spi_loopback/src/spi.c#L15
@kapi-no can you check if there's another difference between yours and the spi one?

Reply:
Well, the spi test project specifies that it requires a platform with SPI capability:

https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/spi/spi_loopback/testcase.yaml#L3

Probably, the SPI test is not generated & run on ```qemu_x86```, because it has no such capability.

To be clear, if I remove the ```platform_whitelist``` attribute, I am still able to generate a valid project for ```qemu_x86``` with cmake. The issue appears when the project is generated in the Zephyr CI environment that is prepared by ```scripts/ci/run_ci.sh```. 

I will try to look into this when I have some spare time, but I suspect that it is some kind of general issue for all tests that run on ```qemu_x86```.

Reply:
I investigated this a bit more and the tests fail actually on all platforms except ```qemu_x86``` in the current state. This is because there is only one ```prj_qemu_x86.conf``` file that is used for ```qemu_x86```. Other platforms use default configuration which does not include Ztest framework.

However, when I defined a default configuration for all platforms using ```prj.conf```, the Zephyr CI failed during the previous rebase:
https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/51146/4/console
Various platforms lacked necessary DT definitions (e.g. ```DT_FLASH_AREA_STORAGE_ID``` / ```DT_FLASH_ERASE_BLOCK_SIZE``` for ```qemu_riscv32```, ```qemu_riscv64``` or ```qemu_x86_64```)

I question the need to have the settings NVS backend tests run on all platforms (192 more tests). These tests are supposed to verify the Settings module logic - not the underlying flash drivers.

To sum up, I think it's better to use ```platform_whitelist``` attribute and have single test executed for ```qemu_x86```. Do you have any objections?

Reply:
Yes, I still object to the whitelist. You can add `depends_on: nvs` to filter out boards that don't have the necessary DT definitions. See `samples/subsys/nvs/sample.yaml`

```
$ git grep nvs -- boards/*/*/*.yaml
boards/arm/bl652_dvk/bl652_dvk.yaml:  - nvs
boards/arm/disco_l475_iot1/disco_l475_iot1.yaml:  - nvs
boards/arm/efm32hg_slstk3400a/efm32hg_slstk3400a.yaml:  - nvs
boards/arm/efm32pg_stk3402a/efm32pg_stk3402a.yaml:  - nvs
boards/arm/efm32wg_stk3800/efm32wg_stk3800.yaml:  - nvs
boards/arm/efr32_slwstk6061a/efr32_slwstk6061a.yaml:  - nvs
boards/arm/efr32mg_sltb004a/efr32mg_sltb004a.yaml:  - nvs
boards/arm/frdm_k64f/frdm_k64f.yaml:  - nvs
boards/arm/frdm_k82f/frdm_k82f.yaml:  - nvs
boards/arm/nrf51_pca10028/nrf51_pca10028.yaml:  - nvs
boards/arm/nrf51_pca10031/nrf51_pca10031.yaml:  - nvs
boards/arm/nrf52_pca10040/nrf52_pca10040.yaml:  - nvs
boards/arm/nucleo_f091rc/nucleo_f091rc.yaml:  - nvs
boards/arm/nucleo_f334r8/nucleo_f334r8.yaml:  - nvs
boards/arm/nucleo_l432kc/nucleo_l432kc.yaml:  - nvs
boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.yaml:  - nvs
boards/arm/sensortile_box/sensortile_box.yaml:  - nvs
boards/arm/stm32f3_disco/stm32f3_disco.yaml:  - nvs
```

Reply:
@kapi-no  I'm in line with you - this test is for testing whether software correctly uses other software and SUT logic.

I vote for keeping the white-list here.

Reply:
I added NVS capability to the ```qemu_x86``` board so that we can run the test included in this PR on the  ```qemu_x86``` board using:
```depends on: nvs```
syntax instead of:
```platform_whitelist ```

I was forced to add separate ```prj_qemu_x86.conf``` to the ```samples/subsys/nvs/``` sample due to the new NVS capability of ```qemu_x86``` board.

Reply:
@kapi-no, nice work.

I think you should remove prj_qemu_x86.conf from the nvs sample, the sample uses reboot, which is not supported by qemu_x86.

Reply:
Thanks for removing the whitelist. You probably want to add a min_ram property to the testcase.yaml to filter out platforms that don't have enough ram. This will help shippable to pass. For example:
https://github.com/zephyrproject-rtos/zephyr/blob/e833fafd4ba3e7967b32a0c7226d692b48eb5728/tests/net/all/testcase.yaml#L4-L7

Reply:
NVS tests for following boards have failed:
- ```out-3nd-pass/efm32hg_slstk3400a/tests/subsys/settings/nvs/raw/system.settings.nvs```
- ```out-3nd-pass/nucleo_f334r8/tests/subsys/settings/nvs/raw/system.settings.nvs```
because of SRAM overflow
- ```out-3nd-pass/qemu_x86/samples/subsys/nvs/sample.subsys.nvs```
because of timeout. This is sample and I think that framework expects log output in the test format.

```qemu_x86``` is a special type of board, because it is not only compiled but also tested in the runtime. I guess it cannot declare capabilities that are used in samples, because we will get CI failures.

I think we should stick to ```platform_whitelist``` in this case. @MaureenHelm, would you agree to that?

The topic of this PR is not to change how the tests for settings are defined. Apart from that, it is a bit inconsistent to have one settings test using ```depends on``` syntax and other ones using ```platform whitelist``` (fcb, nffs, functional, etc.).

Reply:
@Laczen, I cannot do it since the sample would be built in the default configuration (prj.conf) for ```qemu_x86``` and then the Zephyr CI will fail during compilation. However, I agree with you that we should get rid of ```qemu_x86``` target for the NVS sample.

@MaureenHelm, thanks for the tips on how to solve SRAM overflow issue, but I will still have one failing test:
```out-3nd-pass/qemu_x86/samples/subsys/nvs/sample.subsys.nvs```
This is sample that is now built for  ```qemu_x86``` , because I declared ```nvs``` capability for this board. ```qemu_x86``` is a special type of board, because it is not only compiled but also tested in the runtime. That's why I think that framework expects log output in the test format. Is it possible to somehow blacklist ```qemu_x86``` in the NVS sample?

Reply:
@MaureenHelm, please provide a suggestion on how to solve this last issue so that you can approve this PR. If there is no easy way to solve this, I am planning to use ```platform_whitelist``` attribute.

Reply:
@kapi-no, is it not possible to add depends on: reboot in the nvs sample ?

Reply:
@Laczen, I think there is no such capability

Reply:
You can use `platform_exclude` to blacklist `qemu_x86` in the nvs sample

Reply:
@MaureenHelm, thanks. This is resolved now